convert2rhel/toolopts.py

```diff
@@ -338,5 +388,52 @@ def warn_on_unsupported_options():
         utils.ask_to_continue()
 
 
+def options_from_config_files(cfg_path=None):
+    """Get the options from config file.
+
+    The order is:
+    1. if filename is set then load from it
+    2. ~/.convert2rhel.ini
+    3. /etc/convert2rhel.ini
+
+    Return dict of supported options with their values.
+    """
+    headers = ["subscription_manager"]  # supported sections in config file
+    config_file = configparser.ConfigParser()
+    paths = [os.path.expanduser(p) for p in CONFIG_PATHS]
+    # Create dict with all supported options, all of them set to None
+    # needed for avoiding problems with files priority
+    # The name of supported option MUST correspond with the name in ToolOpts()
+    # Otherwise it won't be used
+    supported_opts = {"password": None, "activation_key": None}
+
+    if cfg_path:
+        cfg_path = os.path.expanduser(cfg_path)
+        if not os.path.exists(cfg_path):
+            raise OSError(2, "No such file or directory: '" + cfg_path + "'")
+        paths.insert(0, cfg_path)  # highest priority
+
+    for path in paths:
+        if os.path.exists(path):
+            if not oct(os.stat(path).st_mode)[-4:].endswith("00"):
+                loggerinst.critical("The %s file must only be accessible by the owner (0600)" % path)
+            config_file.read(path)
+
+            for header in config_file.sections():
+                if header in headers:
+                    for option in config_file.options(header):
+                        if option.lower() in supported_opts:
+                            # Solving priority
+                            if supported_opts[option.lower()] is None:
+                                supported_opts[option] = config_file.get(header, option)
+                                loggerinst.debug("Found %s in %s" % (option, path))
+                        else:
+                            loggerinst.warning("Unsupported option %s in %s" % (option, path))
+                elif header not in headers and header != "DEFAULT":
+                    loggerinst.warning("Unsupported header %s in %s." % (header, path))
```

Comment:
I'm wondering, should we include an `else` statement by the end to log that the configuration file was not found? Something like: 
```suggestion
                elif header not in headers and header != "DEFAULT":
                    loggerinst.warning("Unsupported header %s in %s." % (header, path))
        else:
            loggerinst.debug("Couldn't find configuration file at '%s'." % path)
```

It feels to me that this could help us in debugging customer logs if they ever report that convert2rhel couldn't read the configuration the way it should. 

Let me know what you think about it, @hosekadam and @abadger